test/integration/local/test_serving.py

```diff
@@ -1,21 +1,49 @@
+# Copyright 2017-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
 from __future__ import absolute_import
 
 import os
 
-import requests
+from sagemaker.chainer import ChainerModel
+from sagemaker.predictor import BytesDeserializer
 
-from test.utils import local_mode
+from test.utils import test_utils
 
 path = os.path.dirname(os.path.realpath(__file__))
-resources_path = os.path.abspath(os.path.join(path, '..', '..', 'resources'))
+resources_path = os.path.abspath(os.path.join(path, '..', '..', 'resources', 'serving'))
 
 
-def test_serving_calls_model_fn_once(docker_image, opt_ml):
+def test_serving_calls_model_fn_once(docker_image, sagemaker_local_session):
     script_path = os.path.join(resources_path, 'call_model_fn_once.py')
-    with local_mode.serve(script_path, model_dir=None, image_name=docker_image, opt_ml=opt_ml,
-                          additional_env_vars=['SAGEMAKER_MODEL_SERVER_WORKERS=2']):
+    model_path = 'file://{}'.format(os.path.join(resources_path, 'model.tar.gz'))
 
-        # call enough times to ensure multiple requests to a worker
-        for i in range(3):
-            # will return 500 error if model_fn called during request handling
-            assert b'output' == requests.post(local_mode.REQUEST_URL, data=b'input').content
+    model = ChainerModel(model_path,
+                         'SageMakerRole',
```

Comment:
Is the role used in localmode? would this work if this role doesn't exist.

Reply:
it's not used - see https://github.com/aws/sagemaker-python-sdk/issues/704